plugins/janus_videoroom.c

```diff
@@ -944,21 +969,19 @@ static void janus_videoroom_notify_participants(janus_videoroom_publisher *parti
 }
 
 static void janus_videoroom_leave_or_unpublish(janus_videoroom_publisher *participant, gboolean is_leaving) {
+	/* we need to check if the room still exists, may have been destroyed already */
+	if(!participant->room || participant->room->destroyed)
+		return;
 	json_t *event = json_object();
+	janus_mutex_lock(&participant->room->mutex);
```

Comment:
Maybe move this lock before the call to `json_object()`? Just to be (more) sure no destruction happens right after our destroyed check.